osu.Game.Rulesets.Osu.Tests/TestSceneOsuBeatSnapGrid.cs

```diff
@@ -0,0 +1,210 @@
+
+        [Cached]
+        private readonly BindableBeatDivisor beatDivisor = new BindableBeatDivisor();
+
+        private TestOsuBeatSnapGrid grid;
+
+        public TestSceneOsuBeatSnapGrid()
+        {
+            editorBeatmap = new EditorBeatmap<OsuHitObject>(new OsuBeatmap());
+
+            createGrid();
+        }
+
+        [SetUp]
+        public void Setup() => Schedule(() =>
+        {
+            Clear();
+
+            editorBeatmap.BeatmapInfo.BaseDifficulty.SliderMultiplier = 1;
+            editorBeatmap.ControlPointInfo.DifficultyPoints.Clear();
+            editorBeatmap.ControlPointInfo.TimingPoints.Clear();
+            editorBeatmap.ControlPointInfo.TimingPoints.Add(new TimingControlPoint { BeatLength = beat_length });
+
+            beatDivisor.Value = 1;
+        });
+
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        [TestCase(4)]
+        [TestCase(6)]
+        [TestCase(8)]
+        [TestCase(12)]
+        [TestCase(16)]
+        public void TestBeatDivisor(int divisor)
+        {
+            AddStep($"set beat divisor = {divisor}", () => beatDivisor.Value = divisor);
+            createGrid();
+        }
+
+        [TestCase(100, 100)]
+        [TestCase(200, 100)]
+        public void TestBeatLength(float beatLength, float expectedSpacing)
+        {
+            AddStep($"set beat length = {beatLength}", () =>
+            {
+                editorBeatmap.ControlPointInfo.TimingPoints.Clear();
+                editorBeatmap.ControlPointInfo.TimingPoints.Add(new TimingControlPoint { BeatLength = beatLength });
+            });
+
+            createGrid();
+            AddAssert($"spacing = {expectedSpacing}", () => Precision.AlmostEquals(expectedSpacing, grid.DistanceSpacing));
+        }
+
+        [TestCase(0.5f, 50)]
+        [TestCase(1, 100)]
+        [TestCase(1.5f, 150)]
+        public void TestSpeedMultiplier(float multiplier, float expectedSpacing)
+        {
+            AddStep($"set speed multiplier = {multiplier}", () =>
+            {
+                editorBeatmap.ControlPointInfo.DifficultyPoints.Clear();
+                editorBeatmap.ControlPointInfo.DifficultyPoints.Add(new DifficultyControlPoint { SpeedMultiplier = multiplier });
+            });
+
+            createGrid();
+            AddAssert($"spacing = {expectedSpacing}", () => Precision.AlmostEquals(expectedSpacing, grid.DistanceSpacing));
+        }
+
+        [TestCase(0.5f, 50)]
+        [TestCase(1, 100)]
+        [TestCase(1.5f, 150)]
+        public void TestSliderMultiplier(float multiplier, float expectedSpacing)
+        {
+            AddStep($"set speed multiplier = {multiplier}", () => editorBeatmap.BeatmapInfo.BaseDifficulty.SliderMultiplier = multiplier);
+            createGrid();
+            AddAssert($"spacing = {expectedSpacing}", () => Precision.AlmostEquals(expectedSpacing, grid.DistanceSpacing));
+        }
+
+        [Test]
+        public void TestCursorInCentre()
```

Comment:
This test is giving me a consistent crash locally in the visual test runner on the last step (same stack trace as the CI build is reporting for it), which is weird, as all of the others run fine to the end.